shell/artifacts/hex/hex-game.js

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2018 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+defineParticle(({DomParticle}) => {
+  const rootTemplate = `
+    <style>
+      hex-game {
+        --hex-x: #F44336;
+        --hex-y: #2196F3;
+      }
+      hex-game[player=x] {
+        --current-player-color: var(--hex-x);
+      }
+      hex-game[player=y] {
+        --current-player-color: var(--hex-y);
+      }
+      hex-game[can-swap] {
+        --swap-color: var(--current-player-color);
+      }
+
+      /* TODO: Remove this once we switch slots to use shadow dom */
+      hexa-gon > div {
+        width: 100%;
+        height: 100%;
+      }
+      hex-cell {
+        width: 100%;
+        height: 100%;
+        display: block;
+      }
+      hex-game[can-swap] hex-cell:hover,
+      hex-cell:not([player]):hover {
+        background: var(--current-player-color);
+        opacity: 0.5;
+      }
+      hex-cell[player=x] {
+        background: var(--hex-x);
+      }
+      hex-cell[player=y] {
+        background: var(--hex-y);
+      }
+    </style>
+    hex-game <hex-game player$="{{player}}" can-swap$="{{canSwap}}" slotid="board"></hex-game>`;
+  const cellTemplate = `<hex-cell player$="{{move}}" on-click="onCellClick" key="{{key}}"></hex-cell>`;
+
+  class Board {
+    constructor(size) {
+      this._size = size;
+      this._board = [];
+      this._player = 0;
+      this._moves = 0;
+      this._winner = null;
+    }
+
+    get size() {
+      return this._size;
+    }
+
+    get canSwap() {
+      return this._moves == 1;
+    }
+
+    get player() {
+      return this.winner ? null : this._player ? 'x' : 'y';
+    }
+
+    get winner() {
+      return this._winner;
+    }
+
+    _tryFindWinner(x, y, seen) {
```

Comment:
If you want to add tests for this later you could follow what I did in Words, basically add an `index.test.html`, add `test/*.js` files, and add it to `package.json` (see `test-mocha-words` which then also has to be added to the `test` entry there). Obviously this setup could be generalized and moreover with @mmandlis new particle testing bits I think maybe we want to use that to test the particle, and have game logic in separate classes for testing as seems useful.